### ISSUE TITLE
Ecto Error serializing details

### ISSUE THREAD

**Issue by dmarkow** [AgilionApps/ja_serializer#50]
http://jsonapi.org/format/#errors says that the `detail` key should be "a human-readable explanation specific to this occurrence of the problem".

Looking at [their example of an error object](http://jsonapi.org/examples/#error-objects):

```json
{
  "errors": [
    {
      "status": "422",
      "source": { "pointer": "/data/attributes/first-name" },
      "title":  "Invalid Attribute",
      "detail": "First name must contain at least three characters."
    }
  ]
}
```

While the `source` key refers to `first-name`, the `detail` key also includes "First name" instead of just "must contain at least three characters." without mentioning "First name".

Right now, the Ecto Error Serializer doesn't include the attribute names in the detail. Because of this, I have to do some string manipulation on the `source` attribute to pull the actual attribute name. It would be super nice to be able to just display the `detail` keys.

While the JSON API spec isn't perfectly clear about including the attribute name in the detail, I would interpret "human readable" to mean "you should be able to show just the `detail` key to a user and have them know what went wrong."  Thoughts? I can work on a PR but wanted to check in first.

**Comment by hunterboerner** [comment]
I concur.

**Comment by alanpeabody** [comment]
Good catch, lets do it. :+1:

**Comment by alanpeabody** [comment]
Interesting question, I usually display errors in context to the field that had errors so I haven't really thought about it.

**Comment by cjbell** [comment]
@alanpeabody this can be closed now #53 is merged

**Comment by alanpeabody** [comment]
thanks!
